crates/ibc/src/core/context.rs

```diff
@@ -534,14 +519,14 @@ pub trait ExecutionContext: ValidationContext {
     /// Stores the given connection_end at path
     fn store_connection(
         &mut self,
-        connections_path: ConnectionsPath,
+        connection_path: ConnectionPath,
```

Comment:
Contrary to what was stated in #319, I think all path arguments should be passed by reference, since either the host will generate the path string with `to_string()`, or read the internal fields to access its store data structure.

Reply:
Because of the recent change, I was hesitant to say that :). My thought was there should be a reservation. 
But yes, I also think so

Reply:
[4babf0f](https://github.com/cosmos/ibc-rs/pull/419/commits/4babf0f95c141908dc3f8fd6b9579cd343fd31fe)